src/ListenerProxy.php

```diff
@@ -44,7 +44,11 @@ public function __construct(OrderedProviderInterface $provider, string $serviceN
      */
     public function addListener(string $methodName, int $priority = 0, string $id = null, string $type = null): string
     {
-        $type = $type ?? $this->getParameterType([$this->serviceClass, $methodName]);
+        try {
+            $type = $type ?? $this->getParameterType([$this->serviceClass, $methodName]);
+        } catch (\InvalidArgumentException $exception) {
+            throw InvalidTypeException::fromClassCallable($this->serviceClass, $methodName, $exception);
+        }
```

Comment:
One last question; why is this repeated here, instead of handed off to a utility method as in OrderedListenerProvider?  It should be the same utility, no?  (Maybe that function goes to a trait, since then it's easy to share between the two classes?)

Reply:
ListenerProxy already had three nearly identical functions; not knowing the reason for that duplication, I didn't want to refactor needlessly when making changes to them. This would be a good opportunity for cleanup, though.